src/rust/madara/sources/setsuscans/src/lib.rs

```diff
@@ -0,0 +1,62 @@
+#![no_std]
+use aidoku::{
+	error::Result, prelude::*, std::net::HttpMethod, std::net::Request, std::String, std::Vec,
+	Chapter, DeepLink, Filter, Listing, Manga, MangaPageResult, Page,
+};
+
+use madara_template::template;
+
+fn get_data() -> template::MadaraSiteData {
+	let data: template::MadaraSiteData = template::MadaraSiteData {
+		base_url: String::from("https://setsuscans.com"),
+		description_selector: String::from("div.summary_content_wrap div p"),
+		get_manga_id,
+		alt_ajax: true,
+		..Default::default()
+	};
+	data
+}
+
+// SetsuScans keeps its manga data in its src attribute as a base64 encoded string
+// so this function overrides the default get_manga_id function to handle SetsuScans
+fn get_manga_id(manga_id: String, base_url: String, path: String) -> String {
+	let url = base_url + "/" + path.as_str() + "/" + manga_id.as_str();
+	let html = Request::new(url.as_str(), HttpMethod::Get).html();
+	let id_html = html.select("script#wp-manga-js-extra").attr("src").read();
+	let id_html = id_html.replace("data:text/javascript;base64,", "");
+	let decoded_html = base64::decode(id_html).unwrap();
+	let decoded_html = String::from_utf8(decoded_html).unwrap();
+	let id = &decoded_html
+		[decoded_html.find("manga_id").unwrap() + 11..decoded_html.find("\"}").unwrap()];
```

Comment:
It's not usually good to use `unwrap()`. Is this always going to exist for every manga? Is there any way to recover from a failure here (I'm guessing probably not)? It's better practice to do `expect()` so if it does end up failing in the future we can see why, rather than having to dig through the code too much.

Reply:
I used `expect()` at first, but changed to `unwrap()` because I wanted it to be consistent with the existing `get_int_manga_id` function. But, since it's bad practice, I'll switch it back to `expect()`.